tools/src/dbgen/Lib.hs

```diff
@@ -30,52 +35,108 @@ import           Pos.Wallet.Web.ClientTypes
+import           Test.QuickCheck                      (arbitrary, generate)
 import           Rendering
 import           Text.Printf
 
 --
 -- Types
 --
 
-data GenSpec = GenSpec {
-    wallets       :: !Integer
+-- | A simple example of how the configuration looks like.
+_testSpec :: GenSpec
+_testSpec = GenSpec
+    { walletSpec = WalletSpec
+        { accounts = 1
+        , accountSpec = AccountSpec { addresses = 100 }
+        , fakeUtxo = FakeUtxoSpec { amount = 1000, distribution = RangeDistribution { range=100 } }
+        }
+    , wallets = 1
+    }
+
+data GenSpec = GenSpec
+    { wallets    :: !Integer
     -- ^ How many wallets to create
-    , wallet_spec :: WalletSpec
+    , walletSpec :: WalletSpec
     -- ^ The specification for each wallet.
     } deriving (Show, Eq, Generic)
 
-instance Interpret GenSpec
+instance FromJSON GenSpec
+instance ToJSON GenSpec
 
-data WalletSpec = WalletSpec {
-    accounts       :: !Integer
+data WalletSpec = WalletSpec
+    { accounts      :: !Integer
     -- ^ How many accounts to generate
-    , account_spec :: AccountSpec
+    , accountSpec  :: AccountSpec
     -- ^ How specification for each account.
+    , fakeUtxo      :: FakeUtxoSpec
+    -- ^ Configuration for the generation of the fake UTxO.
+    } deriving (Show, Eq, Generic)
+
+instance FromJSON WalletSpec
+instance ToJSON WalletSpec
+
+data AccountSpec = AccountSpec
+    { addresses :: !Integer
+    -- ^ How many addresses to generate.
     } deriving (Show, Eq, Generic)
 
-instance Interpret WalletSpec
+instance FromJSON AccountSpec
+instance ToJSON AccountSpec
 
-data AccountSpec = AccountSpec {
-    addresses :: !Integer
-    -- How many addresses to generate.
+-- TODO(ks): The question here is whether we need to support some other
+-- strategies for distributing money, like maybe using a fixed amount
+-- of `toAddress` to cap the distribution - we have examples where we
+-- have around 80 000 addresses which is a lot and can be intensive?
+-- For now, KISS.
+data FakeUtxoSpec = FakeUtxoSpec
+    { amount       :: !Integer
+    -- ^ How much ADA do we want to distribute in a single address.
+    , distribution :: CoinDistributionSpec
+    -- ^ Choose a @CoinDistributionSpec@.
     } deriving (Show, Eq, Generic)
 
-instance Interpret AccountSpec
+instance FromJSON FakeUtxoSpec
+instance ToJSON FakeUtxoSpec
+
+data CoinDistributionSpec
+    = NoDistribution
+    -- ^ Do not distribute the coins.
+    | RangeDistribution { range :: !Integer }
+    -- ^ Distributes to only XX addresses.
+    -- ^ TODO(adn): For now we KISS, later we can add more type constructors
+    deriving (Show, Eq, Generic)
+
+instance FromJSON CoinDistributionSpec
+instance ToJSON CoinDistributionSpec
+
+{-
+λ> encode NoDistribution
+"{\"tag\":\"NoDistribution\"}"
+λ> encode $ RangeDistribution { range=100 }
+"{\"tag\":\"RangeDistribution\",\"range\":100}"
```

Comment:
I'm not totally sold on the idea of relying on whichever format the `Generic` deriving dictates. I think writing a `FromJSON` and `ToJSON` manual instances is so quick and simple we should go for it. It will allow us to have a proper-looking `config` file, something like:

```
{ "type" : "range"
, "addresses": 100 
}
```

As per my original proposal. 

I'm sad to see Dhall going away, but I can see how avoiding the extra dep and a foreign config format might be a quick win for the scope of the tool.